src/main/java/Job.java

```diff
@@ -43,25 +44,17 @@ public boolean canStart(Date currentTime){
     }
 
     public boolean nodesReached(){
-        return currentNodeCount == requiredNodeCount;
+        return currentNodeCount.get() >= requiredNodeCount;
     }
 
     public boolean isRecurring(){
         return jobInterval != 0;
     }
 
-    public void addNodeCount(){
-        int count= currentNodeCount + 1;
-        if(count <= requiredNodeCount) {
-            this.currentNodeCount = count;
-        }
-    }
-
-    public void subtractNodeCount(){
-        int count= currentNodeCount - 1;
-        if(count >= 0) {
-            this.currentNodeCount =count;
-        }
+    public void addNodeCount(int jobinstance){
+            if(currentNodeCount.get()<requiredNodeCount&&jobinstance==instance){ //results of the same job instance
```

Comment:
Can't you use something like

updateAndGet(IntUnaryOperator updateFunction) or  getAndUpdate(IntUnaryOperator updateFunction)

because after that get current node can be changed 
--